docs/docs/terraform/main.tf

```diff
@@ -0,0 +1,466 @@
+}
+
+resource "aws_security_group_rule" "allow_outbound_ipv6" {
+  count           = length(aws_security_group.allow_outbound)
+  type            = "egress"
+  protocol        = "all"
+  from_port       = 0
+  to_port         = 0
+  prefix_list_ids = []
+  ipv6_cidr_blocks = [
+    "::/0"
+  ]
+  cidr_blocks = []
+  security_group_id = aws_security_group.allow_outbound[count.index].id
+}
+
+data "aws_caller_identity" "current" {}
+
+# neptune things
+
+resource "aws_neptune_cluster_parameter_group" "service" {
+  family = "neptune1" # magic, but currently the only family
+  description = var.service
+  name = var.service
+
+  parameter {
+    name = "neptune_enable_audit_log"
+    value = 1
+  }
+
+  parameter {
+    name = "neptune_enforce_ssl"
+    value = 1
+  }
+}
+
+resource "aws_neptune_subnet_group" "service" {
+  name       = "service"
+  subnet_ids = data.aws_subnet_ids.neptune_contained_herein.ids
+}
+
+resource "aws_neptune_cluster" "service" {
+  count = 1
+  engine = "neptune"
+  cluster_identifier = var.service
+  iam_database_authentication_enabled = true
+  # engine_version default
+  neptune_cluster_parameter_group_name = aws_neptune_cluster_parameter_group.service.name
+  vpc_security_group_ids = [
+    aws_security_group.allow_gremlin[0].id,
+    aws_security_group.allow_outbound[0].id,
+  ]
+  neptune_subnet_group_name = aws_neptune_subnet_group.service.name
+  iam_roles = [
+    aws_iam_role.neptune_load_from_s3.arn
+  ]
+
+  # this seems to be renamed soon https://www.terraform.io/docs/providers/aws/r/neptune_cluster.html#delete_protection
+  deletion_protection = true
+  preferred_maintenance_window = "wed:09:49-wed:10:19"
+  backup_retention_period = 1
+  # waits for the maintenance window, but that may not be what you want
+  apply_immediately = false
+  enable_cloudwatch_logs_exports = [ "audit" ]
+}
+
+resource "aws_neptune_parameter_group" "service" {
+  family = "neptune1" # magic, but currently the only family
+  description = var.service
+  name = var.service
+
+  parameter {
+    apply_method = "immediate"
+    name = "neptune_query_timeout"
+    value = 120000
+  }
+}
+
+resource "aws_neptune_cluster_instance" "service" {
+  count = length(aws_neptune_cluster.service)
+  cluster_identifier = aws_neptune_cluster.service[count.index].id
+  # reuse the cluster identifier, but it could be anything
+  identifier = aws_neptune_cluster.service[count.index].cluster_identifier
+  instance_class = "db.t3.medium"
+  neptune_parameter_group_name = aws_neptune_parameter_group.service.name
+  auto_minor_version_upgrade = true
+}
+
+# Supposing you need a public Neptune endpoint, that doesn't happen directly out of the box.
+# Our solution is this:
+#   * put an NLB (network load balancer) in front of each cluster
+#   * the points at an IP address and TCP port (the Neptune server)
+#   * the Neptune server terminates TLS
+#   * but because the address we connect to is different that what it advertises, we install a special
```

Comment:
nit: I wouldn't say "advertises" here since that makes me think about DNS. Instead I think it might be clearer to talk about the Neptune server's end entity certificate not containing a subject that matches the address being connected to.

```suggestion
#   * but because the address we connect to is not present as a subject in the Neptune TLS certificate, we install a special
```